cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -552,7 +554,7 @@ func StartControllers(s *options.CMServer, kubeconfig *restclient.Config, rootCl
 			glog.Fatalf("Failed to get supported resources from server: %v", err)
 		}
 
-		config := restclient.AddUserAgent(kubeconfig, "generic-garbage-collector")
```

Comment:
> The custom user-agent is gone. Intentional? Doesn't match the commit message.

The default client builder being used actually sets it.  The one we're moving to (SA clients) uses a different identity, so you ought to be able to segment there too.